TST: test_info_preserved_pickle_copy_init failed on s390x with TypeError: data type 'void40' not understood

Example log: https://github.com/astropy/astropy/runs/1949615146?check_suite_focus=true

```
________________ test_info_preserved_pickle_copy_init[ndarray] _________________

mixin_cols = OrderedDict([('i', <Column name='i' dtype='int64' format='{0}' description='d' length=4>
0
1
2
3), ('a', <Column name=..., ('m', NdarrayMixin([(7, b'a'), (8, b'b'), (9, b'c'), (9, b'c')],
             dtype=[('f0', '<i4'), ('f1', 'S1')]))])

    def test_info_preserved_pickle_copy_init(mixin_cols):
        """
        Test copy, pickle, and init from class roundtrip preserve info.  This
        tests not only the mixin classes but a regular column as well.
        """
        def pickle_roundtrip(c):
            return pickle.loads(pickle.dumps(c))
    
        def init_from_class(c):
            return c.__class__(c)
    
        attrs = ('name', 'unit', 'dtype', 'format', 'description', 'meta')
        for colname in ('i', 'm'):
            m = mixin_cols[colname]
            m.info.name = colname
            m.info.format = '{0}'
            m.info.description = 'd'
            m.info.meta = {'a': 1}
            for func in (copy.copy, copy.deepcopy, pickle_roundtrip, init_from_class):
                m2 = func(m)
                for attr in attrs:
                    if (attr != 'dtype'
                            or getattr(m.info.dtype, 'isnative', True)
                            or func in (copy.copy, copy.deepcopy)):
                        original = getattr(m.info, attr)
                    else:
                        # func does not preserve byteorder, check against (native) base type.
                        original = m.info.dtype.name
>                   assert getattr(m2.info, attr) == original
E                   TypeError: data type 'void40' not understood

astropy/table/tests/test_mixin.py:450: TypeError
```

I suspect this is caused by #11294 but maybe @dhomeier , @taldcroft , or @mhvk can confirm.

--- a/astropy/table/tests/conftest.py
+++ b/astropy/table/tests/conftest.py
@@ -146,8 +146,10 @@ MIXIN_COLS = {'quantity': [0, 1, 2, 3] * u.m,
                   10*u.km/u.s),
               'arraywrap': ArrayWrapper([0, 1, 2, 3]),
               'arrayswap': ArrayWrapper(np.arange(4, dtype='i').byteswap().newbyteorder()),
-              'ndarray': np.array([(7, 'a'), (8, 'b'), (9, 'c'), (9, 'c')],
+              'ndarraylil': np.array([(7, 'a'), (8, 'b'), (9, 'c'), (9, 'c')],
                                   dtype='<i4,|S1').view(table.NdarrayMixin),
+              'ndarraybig': np.array([(7, 'a'), (8, 'b'), (9, 'c'), (9, 'c')],
+                                  dtype='>i4,|S1').view(table.NdarrayMixin),
               }
 MIXIN_COLS['earthlocation'] = coordinates.EarthLocation(
     lon=MIXIN_COLS['longitude'], lat=MIXIN_COLS['latitude'],

--- a/astropy/table/tests/test_mixin.py
+++ b/astropy/table/tests/test_mixin.py
@@ -429,8 +429,10 @@ def test_info_preserved_pickle_copy_init(mixin_cols):
         for func in (copy.copy, copy.deepcopy, pickle_roundtrip, init_from_class):
             m2 = func(m)
             for attr in attrs:
+                # non-native byteorder not preserved by last 2 func, _except_ for structured dtype
                 if (attr != 'dtype'
                         or getattr(m.info.dtype, 'isnative', True)
+                        or m.info.dtype.name.startswith('void')
                         or func in (copy.copy, copy.deepcopy)):
                     original = getattr(m.info, attr)
                 else:

--- a/astropy/table/tests/test_mixin.py
+++ b/astropy/table/tests/test_mixin.py
@@ -436,8 +436,8 @@ def test_info_preserved_pickle_copy_init(mixin_cols):
                         or func in (copy.copy, copy.deepcopy)):
                     original = getattr(m.info, attr)
                 else:
-                    # func does not preserve byteorder, check against (native) base type.
-                    original = m.info.dtype.name
+                    # func does not preserve byteorder, check against (native) type.
+                    original = m.info.dtype.newbyteorder('=')
                 assert getattr(m2.info, attr) == original
 
 

--- a/astropy/table/tests/test_table.py
+++ b/astropy/table/tests/test_table.py
@@ -1865,7 +1865,7 @@ class TestPandas:
     def test_mixin_pandas(self):
         t = table.QTable()
         for name in sorted(MIXIN_COLS):
-            if name != 'ndarray':
+            if not name.startswith('ndarray'):
                 t[name] = MIXIN_COLS[name]
 
         t['dt'] = TimeDelta([0, 2, 4, 6], format='sec')

